sql/moz-fx-data-shared-prod/telemetry_derived/urlbar_search_sessions_v1/query.sql

```diff
@@ -0,0 +1,215 @@
+WITH events_unnested AS (
+  SELECT
+    DATE(submission_timestamp) AS submission_date,
+    sample_id,
+    client_info.client_id,
+    timestamp AS event_timestamp,
+    normalized_channel,
+    normalized_country_code,
+    name AS event_name,
+    CASE
+      WHEN name = 'engagement'
+        AND (
+          (
+            mozfun.map.get_key(extra, "engagement_type") IN (
+              "click",
+              "drop_go",
+              "enter",
+              "go_button",
+              "paste_go"
+            )
+          )
+        )
+        THEN 'engaged'
+      WHEN name = 'abandonment'
+        THEN 'abandoned'
+      WHEN name = 'engagement'
+        AND (
+          (
+            mozfun.map.get_key(extra, "engagement_type") NOT IN (
+              "click",
+              "drop_go",
+              "enter",
+              "go_button",
+              "paste_go"
+            )
+          )
+        )
+        THEN "annoyance"
+      ELSE NULL
+    END AS search_session_type,
+    SPLIT(mozfun.map.get_key(extra, "results"), ',')[OFFSET(0)] AS result_type,
+    SPLIT(mozfun.map.get_key(extra, "results"), ',') AS results,
+    mozfun.map.get_key(extra, "selected_result") AS selected_results,
+    mozfun.map.get_key(extra, "engagement_type") AS engagement_type,
+    mozfun.map.get_key(extra, "n_chars") AS number_of_chars_typed,
+    mozfun.map.get_key(extra, "n_results") AS num_total_results,
+    metrics,
+    metrics.uuid.legacy_telemetry_client_id AS legacy_telemetry_client_id,
+    ping_info.experiments
+  FROM
+    `moz-fx-data-shared-prod.firefox_desktop_stable.events_v1`,
+    UNNEST(events)
+  CROSS JOIN
+    UNNEST(ping_info.experiments) AS e
+  WHERE
+    DATE(submission_timestamp) = @submission_date
+    AND category = 'urlbar'
+    AND name IN ('engagement', 'abadonment')
+),
+events_summary AS (
+  SELECT
+    submission_date,
+    sample_id,
+    client_id,
+    event_timestamp,
+    normalized_channel,
+    normalized_country_code,
+    event_name,
+    search_session_type,
+    CASE
+      WHEN search_session_type = 'engaged'
+        THEN engagement_type
+      ELSE NULL
+    END AS engaged_result_type,
+    result_type,
+    number_of_chars_typed,
+    num_total_results,
+    CASE
+      WHEN search_session_type = 'annoyance'
+        THEN engagement_type
+      ELSE NULL
+    END AS annoyance_signal_type,
+    COUNTIF(
+      res IN ('search_suggest', 'search_history', 'search_suggest_rich')
```

Comment:
Noting that if `res` is now `unnest(product_results)` using the UDF, all the COUNTIFs need to be updated.

Reply:
Yes, I have suggested a change.